Source/astcenc_vecmathlib_neon_armv7_4.h

```diff
@@ -0,0 +1,174 @@
+{
+	uint32x4_t amask = vceqq_f32(a, a);
+	uint32x4_t bmask = vceqq_f32(b, b);
+	a = vbslq_f32(amask, a, b);
+	b = vbslq_f32(bmask, b, a);
+	return vminq_f32(a, b);
+}
+
+/**
+ * @brief Return the horizontal maximum of a vector.
+ */
+ASTCENC_SIMD_INLINE float32_t vmaxvq_f32(float32x4_t a)
+{
+	float32_t a0 = vgetq_lane_f32(a, 0);
+	float32_t a1 = vgetq_lane_f32(a, 1);
+	float32_t a2 = vgetq_lane_f32(a, 2);
+	float32_t a3 = vgetq_lane_f32(a, 3);
+	return std::max(std::max(a0, a1), std::max(a2, a3));
+}
+
+/**
+ * @brief Return the horizontal maximum of a vector.
+ */
+ASTCENC_SIMD_INLINE float32_t vminvq_f32(float32x4_t a)
+{
+	float32_t a0 = vgetq_lane_f32(a, 0);
+	float32_t a1 = vgetq_lane_f32(a, 1);
+	float32_t a2 = vgetq_lane_f32(a, 2);
+	float32_t a3 = vgetq_lane_f32(a, 3);
+	return std::min(std::min(a0, a1), std::min(a2, a3));
+}
+
+/**
+ * @brief Return the horizontal maximum of a vector.
+ */
+ASTCENC_SIMD_INLINE int32_t vmaxvq_s32(int32x4_t a)
+{
+	int32_t a0 = vgetq_lane_s32(a, 0);
+	int32_t a1 = vgetq_lane_s32(a, 1);
+	int32_t a2 = vgetq_lane_s32(a, 2);
+	int32_t a3 = vgetq_lane_s32(a, 3);
+	return std::max(std::max(a0, a1), std::max(a2, a3));
+}
+
+/**
+ * @brief Return the horizontal maximum of a vector.
+ */
+ASTCENC_SIMD_INLINE int32_t vminvq_s32(int32x4_t a)
+{
+	int32_t a0 = vgetq_lane_s32(a, 0);
+	int32_t a1 = vgetq_lane_s32(a, 1);
+	int32_t a2 = vgetq_lane_s32(a, 2);
+	int32_t a3 = vgetq_lane_s32(a, 3);
+	return std::min(std::min(a0, a1), std::min(a2, a3));
+}
+
+/**
+ * @brief Return the sqrt of the lanes in the vector.
+ */
+ASTCENC_SIMD_INLINE float32x4_t vsqrtq_f32(float32x4_t a)
+{
+	a = vmaxq_f32(a, vdupq_n_f32((std::numeric_limits<float>::min)()));
+	float32x4_t x = vrsqrteq_f32(a);
+	x = vmulq_f32(vrsqrtsq_f32(a, vmulq_f32(x, x)), x);
+	x = vmulq_f32(vrsqrtsq_f32(a, vmulq_f32(x, x)), x);
+	return vmulq_f32(a, x);
+}
+
+/**
+ * @brief Return a float rounded to the nearest integer value.
+ */
+ASTCENC_SIMD_INLINE float32x4_t vrndnq_f32(float32x4_t a)
+{
+	assert(::fegetround() == FE_TONEAREST);
+	float32_t a0 = ::nearbyintf(vgetq_lane_f32(a, 0));
+	float32_t a1 = ::nearbyintf(vgetq_lane_f32(a, 1));
+	float32_t a2 = ::nearbyintf(vgetq_lane_f32(a, 2));
+	float32_t a3 = ::nearbyintf(vgetq_lane_f32(a, 3));
+	return {a0, a1, a2, a3};
+}
+
+/**
+ * @brief vector by vector division.
+ */
+ASTCENC_SIMD_INLINE float32x4_t vdivq_f32(float32x4_t a, float32x4_t b)
+{
+	float32x4_t x = vrecpeq_f32(b);
+	x = vmulq_f32(vrecpsq_f32(b, x), x);
+	x = vmulq_f32(vrecpsq_f32(b, x), x);
+	return vmulq_f32(a, x);
+}
+
+/**
+ * @brief Table vector Lookup.
+ */
+ASTCENC_SIMD_INLINE int8x16_t vqtbl1q_s8(int8x16_t t, uint8x16_t idx)
+{
+	int8x8x2_t tab;
+	tab.val[0] = vget_low_s8(t);
+	tab.val[1] = vget_high_s8(t);
+	int8x16_t id = vreinterpretq_s8_u8(idx);
+	return vcombine_s8(
+		vtbl2_s8(tab, vget_low_s8(id)),
+		vtbl2_s8(tab, vget_high_s8(id)));
+}
+
+/**
+ * @brief Add across Vector.
+ */
+ASTCENC_SIMD_INLINE uint32_t vaddvq_u32(uint32x4_t a)
```

Comment:
Please remove vaddvq; it's no longer needed as I removed uses of it in the main aarch64 NEON code, and it's impossible to make invariant with x86 SIMD implementations.